dev/breeze/src/airflow_breeze/utils/selective_checks.py

```diff
@@ -606,7 +606,43 @@ def parallel_test_types(self) -> str:
                     )
                     test_types_to_remove.add(test_type)
             current_test_types = current_test_types - test_types_to_remove
-        return " ".join(sorted(current_test_types))
+        for test_type in tuple(current_test_types):
+            if test_type == "Providers":
+                current_test_types.remove(test_type)
+                current_test_types.update(
+                    ("Providers[amazon]", "Providers[google]", "Providers[-amazon,google]")
+                )
+            elif test_type.startswith("Providers[") and "amazon" in test_type or "google" in test_type:
+                current_test_types.remove(test_type)
+                if "amazon" in test_type:
+                    current_test_types.add("Providers[amazon]")
+                if "google" in test_type:
+                    current_test_types.add("Providers[google]")
```

Comment:
this is all pretty specific... Running tests in parallel for all providers regardless is not good ?

Reply:
We cannot run tests in parallell, because far too many of our tests rely on a shared database (for example connections are not mocked. DagRuns are created, etc. etc. ) . Simpy speaking HUGE percent of our tests are not pure unit tests with everything mocked but they rely on a shared database to be there. We even run all our tests WITH specific database. New tests that continue using the shared database are added / modified /updated in every PR.

if we run them in parallel, the tests will start override each other data in the database. 

So we can do either of:

1) review the 12.500 of tests of ours and separate out "real unit test" from the "DB tests" and  add mechanisms to keep the separaiton - then we would be able to parallelise the "real unit tests". Possibly even rewrite the tests to be "real unit tests" and mock the DB access

2)  or do what we are doing - i.e. split the tests into  more-or-less equal chunks (in terms of execution time) and run them sequentially, each of the test with its own database (this is what we do now).

Option 1) seems to require enormous effort - but if you (or anyone) would like to take on the task, this is a  good idea.  I would love to have it, but it seems not feasible (but I would love to be proven wrong)

Option 2) make a deliberate effort to split the tests and balance-optimize them from time to time with few hours effort and custom parallel running framework (this is what we have now)

Option 3.) ... I do not see a 3rd option

But maybe there is one? Courious to hear your thoughts :)

Reply:
BTW. If you want to see what's going on when you try to parallelize tests, I encourage you:

```
1) enter breeze with a lot of CPUs available
2) run pytest -n auto tests
3) Have fun
```

Reply:
And yes. That question is being asked from time to time (including myself) :) . The answer has not changed since the day I joined the project.

Reply:
cc: @vandonr-amz - I just run the `pytest -n auto tests` for the fun of it.  This has always ben an interesting experience (and pretty much always a different one). 

This time It failed pytest collection:

```
2 skipped, 2520 warnings, 407 errors in 313.56s (0:05:13`
```

Most of them:

```
E   botocore.exceptions.ClientError: An error occurred (UnrecognizedClientException) when calling the GetParameter operation: The security token included in the request is invalid.
```

Seems that AWS tests have problems to be even collected in parallel :)

Reply:
BTW. I am really not trying to mock you :D - it's just REALLY our tests are ment to be collected and executed sequentially and it would be HUGE effort to change it, that's why I figured out that more feasible way to utilise multiple CPUs, one that does not cost enormoust amount of engineering time (not mentioning mental breakdown while doing so) is to split the tests in sequential chunks that are run in parallel each with own database and in an isolated container.

This is a way to keep sanity.

Reply:
Oh yeah ok I was not aware of those shortcomings. I'm just a bit weary of introducing some `if amazon:` in the code, but I understand how the situation here is... special ^^

Reply:
It is :)

Reply:
But maybe some day, someone will make it less specific :). I just had a conversation where we discussed how things were working when I joined the project 5 years ago ... and welll.. A LOT has changed since. So maybe in a few year's time :D. Who knows.